### PR TITLE
Restore endpoint to set a breakpoint

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: modules/engine/src/main/scala/observe/engine/Step.scala [hunk 1]
```diff
@@ -24,16 +25,17 @@ final case class Step[F[_]](
...
 object Step {
 
-  final case class BreakpointMark(self: Boolean) extends AnyVal
-
-  final case class SkipMark(self: Boolean) extends AnyVal
-
-  final case class Skipped(self: Boolean) extends AnyVal
+  object BreakpointMark extends NewType[Boolean]
+  type BreakpointMark = BreakpointMark.Type
+  object SkipMark extends NewType[Boolean]
```
Comment (rpiaggio) [inline modules/engine/src/main/scala/observe/engine/Step.scala:30]: I think we decided not to have skipped steps anymore. Maybe we can remove the logic? It can be another PR.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: modules/engine/src/main/scala/observe/engine/Step.scala (hunk 1 above)
Comment (cquiroz) [reply modules/engine/src/main/scala/observe/engine/Step.scala:30]: Ok, I'll do that, there is some leftover code on the engine about them
